lib/Sema/CSDiagnostics.cpp

```diff
@@ -1000,23 +1029,41 @@ bool MemberAccessOnOptionalBaseFailure::diagnoseAsError() {
   auto unwrappedBaseType = baseType->getOptionalObjectType();
   if (!unwrappedBaseType)
     return false;
+  
+  auto baseSourceRange = getMemberBaseSourceRange();
+  
+  emitDiagnosticAt(baseSourceRange.End, diag::optional_base_not_unwrapped,
+                   baseType, Member, unwrappedBaseType);
+
+  auto componentPathElt =
+      locator->getLastElementAs<LocatorPathElt::KeyPathComponent>();
+  if (componentPathElt && componentPathElt->getIndex() == 0) {
+    // For members where the base type is an optional key path root
+    // let's emit a tailored note suggesting to remove the '?' and a
+    // fix to replace the optional type with its unwrapped type.
+    if (auto *nominalDecl = unwrappedBaseType->getAnyNominal()) {
```

Comment:
What if it's not a nominal root, such as `\(Int, Int)?.0`? Rather than digging out a decl, we should be able to print the unwrapped type directly through `unwrappedBaseType.getString()`.